common/src/main/java/com/netflix/conductor/common/metadata/workflow/WorkflowTask.java

```diff
@@ -13,33 +13,60 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/**
- * 
- */
 package com.netflix.conductor.common.metadata.workflow;
 
+import com.github.vmg.protogen.annotations.ProtoEnum;
 import com.github.vmg.protogen.annotations.ProtoField;
 import com.github.vmg.protogen.annotations.ProtoMessage;
 import com.netflix.conductor.common.metadata.tasks.TaskDef;
 
 import java.util.Collection;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Iterator;
 import java.util.LinkedHashMap;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 
 /**
  * @author Viren
  *
- * This is the task definition definied as part of the {@link WorkflowDef}. The tasks definied in the Workflow definition are saved
+ * This is the task definition defined as part of the {@link WorkflowDef}. The tasks definied in the Workflow definition are saved
  * as part of {@link WorkflowDef#tasks}
  */
 @ProtoMessage
 public class WorkflowTask {
 
+	/**
+	 * This field is deprecated and will be removed in the next version.
+	 * Please use {@link TaskType} instead.
+	 */
+	@Deprecated
+    public enum Type {
```

Comment:
How will this fix the issue of backward compatibility if this isn't used anywhere?

Reply:
Overload setType method to take WorkflowTask.Type as input and convert it to TaskType, so workers that create dynamic tasks will work with latest client.

Reply:
overloaded getWorkflowTaskType() method and added @deprecated annotation to it.

Reply:
> overloaded setWorkflowTaskType() method and added @deprecated annotation to it.

i take that back, cannot add an overloaded setter otherwise the object mapper fails